docs/trie.md

```diff
@@ -0,0 +1,90 @@
+# Trie
+
+We use Merkle-Patricia Trie to store blockchain state.
+Trie is persistent, which means that insertion of new node actually leads to creation of new path to this node, and thus root of Trie after insertion will also be presented by new object.
+
+Here we describe its implementation details which are closely related to Runtime.
+
+## Main structures
+
+### Trie
+
+Base structure. 
+It is stored in the RocksDB, which is persistent across node restarts. Trie communicates with database using `TrieStorage`.
+Trie data is stored in key-value format in `ColState` column, wherein nodes are stored altogether with `Vec<u8>` values.
```

Comment:
Not sure what you mean by "stored altogether with `Vec<u8>` values`

Reply:
The point is that on the DB level, we store two kinds of DB records - trie nodes and values. I added more details about it now.